### DISCUSSION TITLE
TLS 1.3 support with 5.0.0.0 .NET v4.0.30319?

### DISCUSSION THREAD

**Discussion by JanBan** [rabbitmq/rabbitmq-dotnet-client#1240] [Q&A] [answered]
Hello, 

We are eyeing support for TLS 1.3 in our RabbitMQ communication through the .NET client, and we are not clear if there is any specific version of the client that is required in order to enable it. 
In the release notes I don't see TLS support explicitly mentioned under any version, docs didn't return anything as well - perhaps I've overlooked. 

Right now we are on 5.0.0.0, though the library was downloaded as binary and Runtime Version specified is v4.0.30319. Should we be good with this one? Framework v4.0.30319 wouldn't support TLS 1.3 AFAIK. How do we get a higher runtime 5.0.0.0 one? Do we have to build it ourselves? 

Thank you!

**Comment by michaelklishin** [comment, marked as answer]
The client does not implement TLS, so as long as the .NET version you run on supports it and you can pass in a constant, it should be fine. That said, I haven't tried it with .NET specifically.

**Reply by JanBan** [reply]
Hi @michaelklishin, thanks for your reply. 
Could the fact that the client binary that we've downloaded is build against .NET v4.0.30319 present any issues? Our code that uses it targets 4.8, so hopefully we should be good to go here to my understanding.

**Reply by michaelklishin** [reply]
It don't think the client ties itself to a particular version when it comes to TLS. It simply doesn't use any new libraries or abstractions or C# features available in .NET 4.8. I'd say unless .NET's TLS implementation has had major breaking changes between those versions, you should be fine.

**Reply by JanBan** [reply]
Okay, I see. Thanks!
